packages/engine/Source/Scene/MetadataPicking.js

```diff
@@ -0,0 +1,308 @@
+    classProperty,
+    dataView,
+    offset
+  );
+  return result;
+};
+
+/**
+ * Decode the given raw values into the raw (array-based) form of
+ * a metadata property value.
+ *
+ * (For decoding to types like `CartesianN`, the `decodeMetadataValues`
+ * function can be used)
+ *
+ * The given values are a `Uint8Array` containing the RGBA
+ * values that have been read from the metadata picking
+ * frame buffer. They are assumed to contain the value for
+ * the given class property, as encoded by the
+ * `MetadataPickingPipelineStage` for metadata picking.
+ *
+ * When the given class property is an array, then (it has to be
+ * a fixed-length array, and) the result will be an array with
+ * the respective length.
+ *
+ * When the given class property is vector- or matrix typed,
+ * then the result will be an array, with a length that corresponds
+ * to the number of vector- or matrix components.
+ *
+ * (The case that the property is an array of vector- or matrix
+ * elements is not supported on the side of the general metadata
+ * shader infrastructure, but handled here nevertheless. For such
+ * an input, the result would be an array of arrays, with each
+ * element representing one of the vectors or matrices).
+ *
+ * In any case, the return value will be the "raw" value, which does
+ * take into account normalization, but does NOT take into account
+ * any offset/scale, or default/noData value handling.
+ *
+ * @param {MetadataClassProperty} classProperty The `MetadataClassProperty`
+ * @param {Uint8Array} rawPixelValues The raw values
+ * @returns {number|bigint|number[]|bigint[]|undefined} The value
+ *
+ * @private
+ */
+MetadataPicking.decodeRawMetadataValues = function (
+  classProperty,
+  rawPixelValues
+) {
+  const dataView = new DataView(
+    rawPixelValues.buffer,
+    rawPixelValues.byteOffset,
+    rawPixelValues.byteLength
+  );
+  if (classProperty.isArray) {
+    const arrayLength = classProperty.arrayLength;
+    const result = Array(arrayLength);
+    for (let i = 0; i < arrayLength; i++) {
+      const element = MetadataPicking.decodeRawMetadataValueElement(
+        classProperty,
+        dataView,
+        i
+      );
+      result[i] = element;
+    }
+    return result;
+  }
+  const result = MetadataPicking.decodeRawMetadataValueElement(
+    classProperty,
+    dataView,
+    0
+  );
+  return result;
+};
+
+/**
+ * Converts the given type into an object representation where appropriate.
+ *
+ * When the given type is `SCALAR`, `STRING`, `BOOLEAN`, or `ENUM`, or
+ * when the given value is `undefined`, then the given value will be
+ * returned.
+ *
+ * Otherwise, for the `VECn/MATn` types, the given value is assumed to be
+ * a numeric array, and is converted into the matching `CartesianN/MatrixN`
+ * value.
+ *
+ * @param {string} type The `ClassProperty` type
+ * @param {number|bigint|number[]|bigint[]|undefined} value The input value
+ * @returns The object representation
```

Comment:
Similar to the above: This could be quite a lot of things. I tried to keep these functions somewhat generic. Now they are, in fact, more generic than they have to be for what is _currently_ supported via the frame-buffer-based picking. (For example, it does handle the `MAT4/Matrix4` case. This cannot be part of a property texture ... for now).

I hope that `{any}`, even though not helpful, is sufficient in _this_ particular case.